src/c_kzg_4844.c

```diff
@@ -1315,26 +1331,46 @@ C_KZG_RET verify_aggregate_kzg_proof(bool *out,
                                      const KZGProof *kzg_aggregated_proof,
                                      const KZGSettings *s) {
     C_KZG_RET ret;
-    Polynomial* polys = calloc(n, sizeof(Polynomial));
-    if (polys == NULL) return C_KZG_MALLOC;
+    g1_t* commitments = NULL;
+    Polynomial* polys = NULL;
+
+    g1_t proof;
+    ret = bytes_to_g1(&proof, (uint8_t *)(kzg_aggregated_proof));
+    if (ret != C_KZG_OK) goto out;
+
+    commitments = calloc(n, sizeof(g1_t));
+    if (commitments == NULL) {
+        ret = C_KZG_MALLOC;
+        goto out;
+    }
+
+    polys = calloc(n, sizeof(Polynomial));
+    if (polys == NULL) {
```

Comment:
We should also be checking `0 < n` here, since `calloc(0, _)` isn't portably specified (e.g., may return `NULL` on success). I can fix it later if you want though.

Reply:
I agree. Fixed. It's a simple change, might as well do it here.